### PR TITLE
impl for get block from orderer

### PR DESCRIPTION
**Pull request by SamYuan1990**
Hope by this PR we can support get block from orderer.

### CODE REVIEW

File: pkg/channel/getBlock.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	"crypto/tls"
+	"fmt"
+
+	"github.com/hyperledger/fabric-admin-sdk/internal/protoutil"
+	"github.com/hyperledger/fabric-admin-sdk/pkg/identity"
+	cb "github.com/hyperledger/fabric-protos-go-apiv2/common"
+	ab "github.com/hyperledger/fabric-protos-go-apiv2/orderer"
+)
+
+var Newest = &ab.SeekPosition{
```
Comment (bestbeforetoday) [inline pkg/channel/getBlock.go:14]: Should be lowercase as not needed in public API

**Review by bestbeforetoday — COMMENTED** [review]
I am not what the purpose is for providing an API the retrieve just the latest block from the orderer. If the intent is to get the current config block, perhaps it would be better to provide an API call that returns the config block, not just the latest block on the blockchain. The config block is obtained by:

1. Getting the latest block
2. Reading the latest config block number from the latest block
3. Getting the latest config block (by block number)

On the publicly exposed API, I don’t think it is helpful to require the caller to understand the Fabric protobuf bindings and create the AtomicBroadcast_DeliverClient themselves. It would be easier for them to just supply a gRPC client connection to the orderer node, and for the admin API implementation to create a AtomicBroadcast_DeliverClient from that gRPC connection, as the chaincode APIs do.

On the implementation, quite a few parameters seem to be passed around between multiple related functions. It might be easier to create a struct with those values as properties, then have the related functions all be receiver functions (or methods) on that struct so the required parameters are implicitly available to all the related functions and don’t need to be explicitly passed around between them.

**Comment by SamYuan1990** [comment]
I will try to update this PR, hope by End of this week.

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]
This change is generally looking pretty good to me. Several suggestions below to:

1. Minimise the API surface to just what is actually required by the client application code; on other words, just getting the config block.
2. Simplify the code required in the client application.
3. Consistency with other parts of the API.

**Review by bestbeforetoday — APPROVED** [review]
Looks good!

**Comment by 1gezhanghao** [comment]
@SamYuan1990 well done, I've tested the 'channel.GetConfigBlockFromOrderer(...)' function and it works great. BTW, could you implement a tool to deep marshal the block into JSON.

**Comment by bestbeforetoday** [comment]
> > @SamYuan1990 well done, I've tested the 'channel.GetConfigBlockFromOrderer(...)' function and it works great. BTW, could you implement a tool to deep marshal the block into JSON.

Note that this is something that the [protolator](https://pkg.go.dev/github.com/hyperledger/fabric-config@v0.1.0/protolator) package in [fabric-tools](https://github.com/hyperledger/fabric-config) does for the old Fabric protobuf (APIv1) Go bindings. There is already issue hyperledger/fabric-config#53 open there to update it for the newer protobuf (APIv2) bindings. I a not against implementing this capability in the fabric-admin-sdk package, but it is important to avoid duplication and co-ordinate across projects. It might be worth getting the opinion of people with an oversight of all the Fabric projects, like @denyeart.

**Comment by denyeart** [comment]
I've always thought that fabric-admin-sdk might be able to use the features that already exist in the fabric-config library, so yes, I'd recommend putting common utilities in fabric-config and using them as a dependency in fabric-admin-sdk.

**Comment by SamYuan1990** [comment]
> I've always thought that fabric-admin-sdk might be able to use the features that already exist in the fabric-config library, so yes, I'd recommend putting common utilities in fabric-config and using them as a dependency in fabric-admin-sdk.

maybe @jt-nti can have a try?
I do agree that we should avoid duplicate works. but, in fabric-config, it seems no end to end test as integrated with fabric as an sample, I am not sure if there is an easy way to use it?(as replace existing code in admin-sdk by fabric-config.)

**Comment by SamYuan1990** [comment]
> I've always thought that fabric-admin-sdk might be able to use the features that already exist in the fabric-config library, so yes, I'd recommend putting common utilities in fabric-config and using them as a dependency in fabric-admin-sdk.

btw, agree with @bestbeforetoday 's comments, fabric-config should better upgrade to protobuf with v2 as admin sdk already made it.

**Pull request merged**